p11-kit/gen-fixed-closures.py

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/python
+
+"""
+SPDX-License-Identifier: BSD-3-Clause
+"""
+
+import io
+import sys
+
+INDENT = "        "
+
+def fixed_instantiate(closures):
+    output = io.StringIO()
+    for i in range(closures):
+        output.write(f"P11_VIRTUAL_FIXED_FUNCTIONS({i})\n")
+
+    initializers_concatenated = ",\n".join([
+        f"{INDENT}P11_VIRTUAL_FIXED_INITIALIZER({i})"
+        for i in range(closures)
+    ])
+    output.write(f'''
+CK_FUNCTION_LIST_3_0 p11_virtual_fixed[P11_VIRTUAL_MAX_FIXED] = {{
+{initializers_concatenated}
+}};
+
+''')
+    for i in range(closures):
+        output.write(f"""\
+P11_VIRTUAL_FIXED_GET_FUNCTION_LIST({i})
+P11_VIRTUAL_FIXED_GET_INTERFACE_LIST({i})
+P11_VIRTUAL_FIXED_GET_INTERFACE({i})
+""")
+    return output.getvalue()
+
+
+if __name__ == "__main__":
+    import argparse
```

Comment:
Any reason why this isn't a global import? Are you planning to support using this as a module?

Reply:
Not for now, but it would make future extension easier I guess.